src/common/file.cpp

```diff
@@ -291,6 +300,14 @@ namespace tools {
       return true;
     }
 
+    fs::path depreciated_path = get_depreciated_default_data_dir();
+    if (fs::is_directory(depreciated_path, ec))
+    {
+      fs::rename(depreciated_path, path);
```

Comment:
This rename could fail, I think, particularly in testnet mode, where we have:
```
default = $HOME/.oxen/testnet
legacy = $HOME/.loki/testnet
```
(the monero code is rather dumb in how it hard codes "/testnet" or "/devnet" onto the data-dir even if you explicitly given it, but oh well).

So if `~/.oxen` doesn't exist at all then we're going to end up trying to rename `~/.loki/testnet` -> `~/.oxen/testnet`, but that will fail because `~/.oxen` doesn't exist; so there needs to be a `create_directories_if_necessary(path.parent_path());` immediately before the rename.

So something like this (untested) in main.cpp:
```C++
// data-dir migration: if not given an explicit data-dir and the default doesn't exist then try to migrate the old one
if (command_line::is_arg_defaulted(vm, cryptonote::arg_data_dir) && !fs::exists(data_dir)) {
  auto old_data_dir = tools::get_depreciated_default_data_dir();
  if (testnet) old_data_dir /= "testnet";
  else if (devnet) old_data_dir /= "devnet";

  if (fs::is_directory(old_data_dir)) {
    if (!tools::create_directories_if_necessary(data_dir.parent_path()))
      throw std::runtime_error{"Data-directory migration failed: can't create " + data_dir.parent_path().u8string()};
    fs::rename(old_data_dir, data_dir);
    fs::create_directory_symlink(data_dir, old_data_dir);
    MGINFO_CYAN("Migrated data directory from " << old_data_dir << " to " << data_dir);
  }
}
```